### PR TITLE
migrating updateCircle backend and frontend to hasura

### LINKED ISSUE #438: Migrate PUT v2/{circle_id}/admin/circles
## Summary:
Migrate Update Circle Laravel API Endpoint functionality to Hasura.

### Details
This task is to migrate the functionality of `PUT v2/{circle_id}/admin/circles` to Hasura.

In Laravel, this endpoint has the following functionality.

**Middleware:** auth token & circle admin

**Additional Validations besides basic inputs:**
n/a

**Logic:**
Checks if vouching setting is turned off. If yes end all nominations for that centre immediately

**DB Tables:**
circles, nominations

### LINKED ISSUE #632: Integrate Hasura Update Circle into FE
## Summary:
Integrate Hasura Update Circle into FE.

### Details
This task is to integrate the frontend of `Update Circle` to Hasura.
https://app.coordinape.com/admin > `Settings` button

**Screenshot of Page**
image

**Logic:**
-

**Related Backend Issue:**
#438

### PR DESCRIPTION
**Pull request by CryptoGraffe**
Fixes #438 and Fixes #632 by migrating updateCircle backend and frontend to hasura. 

Test plan:

1. Use the Admin Circle page to change all the various settings on a circle. Verify that graphql is used in the network tab rather than laravel. 
2. Enable vouching, then nominate some users
3. Edit the circle and disable vouching
4. Check in the DB or the UI that the nominees have ended. 

Also improves zod error handling to include the field that is bad. Try setting an invalid url for the discord webhook to see this.

### CODE REVIEW

File: src/hooks/useApiAdminCircle.ts [hunk 1]
```diff
@@ -17,8 +17,8 @@ export const useApiAdminCircle = (circleId: number) => {
   const { fetchManifest } = useApiBase();
 
   const updateCircle = useRecoilLoadCatch(
-    () => async (params: PutCirclesParam) => {
-      await getApiService().putCircles(circleId, params);
+    () => async (params: ValueTypes['circles_set_input']) => {
+      await mutations.updateCircle(circleId, params);
```
Comment (zashton) [inline src/hooks/useApiAdminCircle.ts:21]: `{"errors":[{"extensions":{"details":[{"path":["input","payload","discord_webhook"],"code":"invalid_string","message":"Invalid url","validation":"url"}],"code":"422"},"message":"Invalid input: Invalid url (discord_webhook)"}]}`

Getting this error when i click on save, i think the issue is the `discord webhook` value is empty before `edit webhook` button is clicked

**Review by zashton — CHANGES_REQUESTED** [review]
image

this button would also need to be migrated, can get the data via

```gql
query MyQuery {
  circles {
    circle_private {
      discord_webhook
    }
  }
}
```
and the save button validation issue would need to be fixed

File: src/hooks/useApiAdminCircle.ts (hunk 1 above)
Comment (CryptoGraffe) [reply src/hooks/useApiAdminCircle.ts]: thanks @zashton this was a good catch. I've changed it to handle it in the way that laravel did, with an `update_webhook` flag. Also handling the empty string as null so we can un-set it.

**Comment by zashton** [comment]
> image
> 
> this button's api call would also need to be migrated, can get the data via
> 
> ```graphql
> query MyQuery {
>   circles {
>     circle_private {
>       discord_webhook
>     }
>   }
> }
> ```
> 
> and the save button validation issue would need to be fixed

you might have missed this comment @CryptoGraffe

**Review by zashton — DISMISSED** [review]
webhook issue fixed!

**Review by topocount — APPROVED** [review]

**Pull request merged**